<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>main_window</class>
 <widget class="QMainWindow" name="main_window">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>847</width>
    <height>563</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>840</width>
    <height>520</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1515541</width>
    <height>515151</height>
   </size>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="acceptDrops">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>Path Finder</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(29, 27, 58);</string>
  </property>
  <property name="animated">
   <bool>true</bool>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="adjust_criteria_button">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>95</width>
      <height>25</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 39, 42);color:white;
</string>
    </property>
    <property name="text">
     <string>Adjust Criteria</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="max_iterations">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>60</y>
      <width>95</width>
      <height>23</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-radius : 8;color:white;</string>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string>50</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="absolute_error_criteria">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>95</width>
      <height>23</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-radius : 8;color:white;</string>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string> 0.00001</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>95</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
    </property>
    <property name="text">
     <string>Epsilon</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>40</y>
      <width>95</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
    </property>
    <property name="text">
     <string>Max Iterations</string>
    </property>
   </widget>
   <widget class="QComboBox" name="selected_method">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>201</width>
      <height>25</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 255, 255);background-color: rgb(70, 72, 110);border-radius : 8;</string>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Bisection</string>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="frame">
     <bool>true</bool>
    </property>
    <item>
     <property name="text">
      <string>Bisection</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>False Position</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Fixed Point</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Newton-Raphson</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Secant</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Bierge-Vieta</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>General Algorithm</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="lower_value">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>95</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
    </property>
    <property name="text">
     <string>Lower Value</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lower_value_text">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>140</y>
      <width>95</width>
      <height>23</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-radius : 8;color:white;</string>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>120</y>
      <width>95</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
    </property>
    <property name="text">
     <string>Upper Value</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="upper_value_text">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>140</y>
      <width>95</width>
      <height>23</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-radius : 8;color:white;</string>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>95</width>
      <height>15</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
    </property>
    <property name="text">
     <string>Notification:</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="notification">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>190</y>
      <width>200</width>
      <height>50</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-radius :15;color:white;</string>
    </property>
   </widget>
   <widget class="MplWidget" name="MplWidget" native="true">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>10</y>
      <width>540</width>
      <height>271</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Adobe Fan Heiti Std B</family>
      <pointsize>7</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>230</width>
      <height>561</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>210</width>
      <height>500</height>
     </size>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color : qlineargradient(spread:pad, x1:0.579, y1:0.210364, x2:0.107, y2:0.988636, stop:0.255682 rgba(25, 26, 44, 255), stop:1 rgba(44, 41, 86, 255))</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="previous_step_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>250</y>
       <width>95</width>
       <height>25</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 39, 42);color:white;
</string>
     </property>
     <property name="text">
      <string>Previous Step</string>
     </property>
    </widget>
    <widget class="QPushButton" name="next_step_button">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>250</y>
       <width>95</width>
       <height>25</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 39, 42);color:white;
</string>
     </property>
     <property name="text">
      <string>Next Step</string>
     </property>
    </widget>
    <widget class="QPushButton" name="edit_function_button">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>90</y>
       <width>95</width>
       <height>25</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 39, 42);color:rgb(255, 249, 250)
</string>
     </property>
     <property name="text">
      <string>Edit Function</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>340</y>
       <width>95</width>
       <height>15</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
     </property>
     <property name="text">
      <string>Step: Lower Value</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="step_lower_value_text">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>360</y>
       <width>201</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;color:white;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>390</y>
       <width>95</width>
       <height>15</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
     </property>
     <property name="text">
      <string>Step: Upper Value</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="step_upper_value_text">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>410</y>
       <width>201</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;color:white;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>490</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
     </property>
     <property name="text">
      <string>Step: Absolute Error</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="step_absolute_error">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>510</y>
       <width>201</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;color:white;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>440</y>
       <width>95</width>
       <height>15</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
     </property>
     <property name="text">
      <string>Step: Root Value</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="step_root_value_text">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>460</y>
       <width>201</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;color:white;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLineEdit" name="iteration_couner_text">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>310</y>
       <width>201</width>
       <height>23</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;color:white;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>95</width>
       <height>15</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(60, 189, 228);background-color : transparent;
</string>
     </property>
     <property name="text">
      <string>Step: Iteration</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="graph_adjustments">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>300</y>
      <width>540</width>
      <height>40</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:white;</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>360</y>
      <width>541</width>
      <height>161</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:white;</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="absolute_error">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:black;background-color : rgb(255, 52, 55);border-radius : 8;
</string>
     </property>
     <property name="text">
      <string>    Absolute Error:</string>
     </property>
    </widget>
    <widget class="QLabel" name="iterations">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>45</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:black;background-color : rgb(255, 52, 55);border-radius : 8;
</string>
     </property>
     <property name="text">
      <string>    Iterations:</string>
     </property>
    </widget>
    <widget class="QLabel" name="root_value">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:black;background-color : rgb(255, 52, 55);border-radius : 8;
</string>
     </property>
     <property name="text">
      <string>    Root Value:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="solve_button">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>110</y>
       <width>90</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 39, 42);color:white;
</string>
     </property>
     <property name="text">
      <string>Solve</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="root_value_text">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>180</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>_____</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="iterations_text">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>45</y>
       <width>180</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>_____</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="absolute_error_text">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>80</y>
       <width>180</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>_____</string>
     </property>
    </widget>
    <widget class="QLabel" name="execution_time">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>115</y>
       <width>90</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:black;background-color : rgb(255, 52, 55);border-radius : 8;
</string>
     </property>
     <property name="text">
      <string>   Execution time:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="execution_time_text">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>115</y>
       <width>180</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>_____</string>
     </property>
    </widget>
    <widget class="QLabel" name="T">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>10</y>
       <width>101</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:black;background-color : rgb(255, 52, 55);border-radius : 8;
</string>
     </property>
     <property name="text">
      <string>  Theoretical Bound</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="theoretical_bound_text">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>10</y>
       <width>101</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>_____</string>
     </property>
    </widget>
    <widget class="QPushButton" name="compare_button">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>60</y>
       <width>90</width>
       <height>41</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 39, 42);color:white;
</string>
     </property>
     <property name="text">
      <string>Compare</string>
     </property>
    </widget>
    <widget class="QLabel" name="T_2">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>60</y>
       <width>101</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color:black;background-color : rgb(255, 52, 55);border-radius : 8;
</string>
     </property>
     <property name="text">
      <string>  Graph Scale</string>
     </property>
    </widget>
    <widget class="QSlider" name="scale_slider">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>90</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="scale_text">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>120</y>
       <width>101</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-radius : 8;</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>_____</string>
     </property>
    </widget>
   </widget>
   <zorder>frame_2</zorder>
   <zorder>frame</zorder>
   <zorder>adjust_criteria_button</zorder>
   <zorder>max_iterations</zorder>
   <zorder>absolute_error_criteria</zorder>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>selected_method</zorder>
   <zorder>lower_value</zorder>
   <zorder>lower_value_text</zorder>
   <zorder>label_4</zorder>
   <zorder>upper_value_text</zorder>
   <zorder>label_5</zorder>
   <zorder>notification</zorder>
   <zorder>MplWidget</zorder>
   <zorder>graph_adjustments</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>847</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MplWidget</class>
   <extends>QWidget</extends>
   <header>mplwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
